vsphere/datadog_checks/vsphere/legacy/vsphere_legacy.py

```diff
@@ -976,6 +978,13 @@ def collect_metrics(self, instance):
             self.gauge('vsphere.vm.count', vm_count, tags=tags)
 
     def check(self, instance):
+        if not self._has_logged_depreciation_warning:
+            self.warning(
+                "DEPRECATION NOTICE: You are using a deprecated version of the vSphere integration. "
+                "To use the newer version, please update your configuration file based on the provided example. "
+                "Look for the `use_legacy_check_version` configuration option."
+            )
+            self._has_logged_depreciation_warning = True
```

Comment:
Why not just put that in `__init__` if we want it to run just once?

Reply:
Yeah that was my first idea, but it doesn't work for some reason. I tried with this simple custom check
```python
class HelloCheck(AgentCheck):
    def __init__(self, *args, **kwargs):
        super(HelloCheck, self).__init__(*args, **kwargs)
        self.warning("This is a warning")
    def check(self, _):
        pass
```,
the warning is logged only once and appear in the status page, but on the next check run, the warning disappears.

Then I tried the following
```python
class HelloCheck(AgentCheck):
    flag = False
    def check(self, _):
        if not HelloCheck.flag:
            self.warning("This is a warning")
            flag = True
```

This also logs the warning a single time, but this time the warning is kept in the status of the agent. TBH I'm not sure why, but this is similar to how we log the warning for the base class.